Tests/SwiftIfConfigTest/Assertions.swift

```diff
@@ -0,0 +1,153 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftDiagnostics
+import SwiftIfConfig
+import SwiftParser
+import SwiftSyntax
+@_spi(XCTestFailureLocation) import SwiftSyntaxMacrosGenericTestSupport
+import XCTest
+import _SwiftSyntaxGenericTestSupport
+import _SwiftSyntaxTestSupport
+
+/// Assert the results of evaluating the condition within an `#if` against the
+/// given build configuration.
+func assertIfConfig(
+  _ condition: ExprSyntax,
+  _ expectedState: ConfiguredRegionState?,
+  configuration: some BuildConfiguration = TestingBuildConfiguration(),
+  diagnostics expectedDiagnostics: [DiagnosticSpec] = [],
+  file: StaticString = #filePath,
+  line: UInt = #line
+) {
+  // Evaluate the condition to check the state.
+  var actualDiagnostics: [Diagnostic] = []
+  do {
+    let actualState = try ConfiguredRegionState(condition: condition, configuration: configuration) { diag in
+      actualDiagnostics.append(diag)
+    }
+    XCTAssertEqual(actualState, expectedState, file: file, line: line)
+  } catch {
+    XCTAssertNil(expectedState, file: file, line: line)
+  }
+
+  // Check the diagnostics.
+  if actualDiagnostics.count != expectedDiagnostics.count {
+    XCTFail(
+      """
+      Expected \(expectedDiagnostics.count) diagnostics, but got \(actualDiagnostics.count):
+      \(actualDiagnostics.map(\.debugDescription).joined(separator: "\n"))
+      """,
+      file: file,
+      line: line
+    )
+  } else {
+    for (actualDiag, expectedDiag) in zip(actualDiagnostics, expectedDiagnostics) {
+      assertDiagnostic(
+        actualDiag,
+        in: .tree(condition),
+        expected: expectedDiag,
+        failureHandler: {
+          XCTFail($0.message, file: $0.location.staticFilePath, line: $0.location.unsignedLine)
+        }
+      )
+    }
+  }
+}
+
+/// Assert that the various marked positions in the source code have the
+/// expected active states.
+func assertActiveCode(
+  _ markedSource: String,
+  configuration: some BuildConfiguration = TestingBuildConfiguration(),
+  states: [String: ConfiguredRegionState],
+  file: StaticString = #filePath,
+  line: UInt = #line
+) throws {
```

Comment:
This is only used in ActiveRegionTests.swift. What do you think about moving it there (and making it `fileprivate`) so that the assertion function is located right next to the tests, which means you don’t need to switch files to see what the assertion does.

I think the same applies to the other assertion functions as well.

Reply:
Sure.